packages/flutter/lib/src/widgets/text_editing_action.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'actions.dart';
+import 'editable_text.dart';
+import 'focus_manager.dart';
+import 'framework.dart';
+
+// Similar to CallbackAction's OnInvokeCallback, but includes EditableTextState.
+typedef _OnInvokeTextEditingCallback<T extends Intent> = Object? Function(T intent, EditableTextState editableTextState);
+
+/// An [Action] related to editing text.
+///
+/// If an [EditableText] is currently focused, then _editableTextState will be
+/// set and the given [onInvoke] callback will be called with it. If not, then
+/// [isEnabled] will be false and [onInvoke] will not be called.
+///
+/// The focused [EditableText] must have a [Key]. This is handled automatically
+/// by built-in text editing widgets like [TextField], [CupertinoTextField],
+/// and [SelectableText], but for custom usage of [EditableText], it is
+/// necessary to explicitly pass in a key.
+///
+/// See also:
+///
+///  * [CallbackAction], which is a similar Action type but unrelated to text
+///    editing.
+class TextEditingAction<T extends Intent> extends Action<T> {
```

Comment:
It might make sense to make TextAction a ContextAction, since eventually I think we're going to need to discover if the _current window_ has the focus (that's an EditableText) rather than just checking primaryFocus, which I assume will correspond to all of the app's windows.

Reply:
Good call, that will probably save me a breaking change in the future 👍